src/sentry/integrations/jira/models/schema_model.py

```diff
@@ -0,0 +1,174 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+
+class JiraSchemaTypes(str, Enum):
+    string = "string"
+    option = "option"
+    array = "array"
+    user = "user"
+    issue_type = "issuetype"
+    issue_link = "issuelink"
+    project = "project"
+    date = "date"
+    team = "team"
+    number = "number"
+    any = "any"
+
+
+@dataclass(frozen=True)
+class JiraSchema:
+    type: str
+    """
+    The Field type. Possible types include:
+    - string
+    - array (has a corresponding `items` field with its subtype)
+    - user
+    - issuetype
+    - issuelink
+    - project (and PROJECT)
+    - date
+    - team
+    - any
+    """
+    custom: str | None = None
+    """
+    The very long custom field name corresponding to some namespace, plugin,
+    and custom field name.
+    """
+    custom_id: str | None = None
+    """
+    A unique identifier for a field on an issue, in the form of 'customfield_<int>'
+    """
+    system: str | None = None
+    """
+    TODO(Gabe): Figure out what this is used for
+    """
+    items: str | None = None
+    """
+    Specifies the subtype for aggregate type, such as `array`. For any
+    non-aggregate types, this will be None
+    """
+
+    __jira_schema_parameter_map = frozenset(
+        [
+            ("type", "type"),
+            ("custom", "custom"),
+            ("custom_id", "custom_id"),
+            ("system", "system"),
+            ("items", "items"),
+        ]
+    )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "JiraSchema":
+        mapped_field_params = {
+            new_name: data.get(new_name)
+            for old_name, new_name in cls.__jira_schema_parameter_map
+            if old_name in data
+        }
+
+        return cls(**mapped_field_params)
+
+    @classmethod
+    def from_dict_list(cls, data: list[dict[str, Any]]) -> list["JiraSchema"]:
+        return [cls.from_dict(item) for item in data]
+
```

Comment:
I'm not super sold on this approach, but figured it'd be nice to have a simplified means of converting from camelCase to snake_case. It also provides some parameter filtering, since we don't use all of the fields Jira provides us on these schemas, especially some of the optional custom fields.